### DISCUSSION TITLE
A Sampler for Graph Matching

### DISCUSSION THREAD

**Discussion by nikita-petrashen** [pyg-team/pytorch_geometric#2263] [Ideas]
Hey, I noticed that there is no mechanism of sampling subgraphs for the task of graph matching. I wrote a snippet myself for this purpose. Is there a need for such a feature? If so, I will try to align it with your guidelines and contribute.

Thank you.

**Comment by rusty1s** [comment]
Can you give a bit more background on the use-case and implementation? You can also just send a PR and I will have a closer look.

**Reply by nikita-petrashen** [reply]
I did some experiments with the DBP15k dataset and the DGMC model. Instead of regular convolutions I tried using hyperbolic ones. This caused memory overflow on a GPU with 16gb memory. So I decided to try training in a mini-batch fashion.

**Reply by rusty1s** [reply]
Sounds good. Can you send a PR so I can have a closer look?

**Reply by nikita-petrashen** [reply]
Sure. This is my first attempt in contributing to github, so it will take some time :) Probably next week.
